### PR TITLE
Update `typing_extensions` minimum + add overrides to `ParallelStrategy`

### PR DESCRIPTION
**Pull request by seanbethard**
Part of #18695 

Updated `typing_extensions` requirement.

`ParallelStrategy` and `SingleDeviceStrategy` inherit from `Strategy`. `SingleDeviceStrategy` defines abstract methods only.

----
:books: Documentation preview :books:: https://pytorch-lightning--18866.org.readthedocs.build/en/18866/

### CODE REVIEW

**Review by Borda — APPROVED** [review]

**Comment by seanbethard** [comment]
I didn't see anything to add for fabric `Strategy`, fabric` Logger`, or `LayerSync`.

in progress:
*  fabric `Accelerator`
* `LightningDataModule`
* `Callback`
* `ClusterEnvironment`
* `Checkpoint IO`

**Comment by awaelchli** [comment]
Thanks for starting this @seanbethard 🎉 
Before you do much more work, we should perhaps discuss where this should be applied in the first place. Could you check my comments (and @carmocca too).

**Comment by seanbethard** [comment]
I think that's everything. please let me know if I missed anything. happy to remove overrides from tests.

**Comment by carmocca** [comment]
@seanbethard I agree with Adrian's comment. I also posted it to the issue: https://github.com/Lightning-AI/lightning/issues/18695#issuecomment-1781555531

Also, please do not update too many files in this PR or else it will be harder to review and get merged. It's better to do it in smaller pieces, plus you get the advantage of extra PRs merged!

**Review by carmocca — COMMENTED** [review]
Every file that uses `@override` needs to import it. You can see the CI failing jobs to find the missing ones

File: src/lightning/fabric/accelerators/cpu.py [hunk 1]
```diff
@@ -62,6 +62,7 @@ def is_available() -> bool:
         """CPU is always available for execution."""
         return True
 
+    @override
```
Comment (awaelchli) [inline src/lightning/fabric/accelerators/cpu.py:65]: This is duplicated

File: src/lightning/pytorch/demos/boring_classes.py [hunk 2]
```diff
@@ -19,6 +19,7 @@
 from torch import Tensor
```
Comment (awaelchli) [inline src/lightning/pytorch/demos/boring_classes.py:19]: we said we would exclude examples

**Comment by awaelchli** [comment]
@Borda This PR no longer contains any relevant changes. I recommend closing it.

**Comment by Borda** [comment]
following @awaelchli suggestion but @seanbethard really pappy and thankful for your contributions

File: src/lightning/pytorch/demos/boring_classes.py (hunk 2 above)
Comment (seanbethard) [reply src/lightning/pytorch/demos/boring_classes.py:19]: you did later qualify after I started that you didn't want overrides in examples. guess I missed this `demos` directory when I removed them from 14 other files here: https://github.com/Lightning-AI/pytorch-lightning/pull/18866/commits/404514cd2bc47ebbf6970b77b53b920d1be8b2f4. sorry about that!

**Comment by seanbethard** [comment]
lol! wow! you're welcome? what contributions? guess I should have signed my commits 😂

File: src/lightning/pytorch/demos/boring_classes.py (hunk 2 above)
Comment (seanbethard) [reply src/lightning/pytorch/demos/boring_classes.py:19]: sorry, I meant 25 files.

**Pull request closed without merging**